src/main/java/htsjdk/samtools/SamInputResource.java

```diff
@@ -281,8 +286,27 @@ public SeekableStream make() {
 
 
     PathInputResource(final Path pathResource) {
+        this(pathResource, Function.identity());
+    }
+
+    PathInputResource(final Path pathResource, Function<SeekableByteChannel, SeekableByteChannel> wrapper) {
         super(Type.PATH);
         this.pathResource = pathResource;
+        this.wrapper = wrapper;
+    }
+
+    /** Returns a modified PathInputResource with the specific wrapper. After calling this,
+     * do NOT use the original object anymore.
+     *
+     * @param wrapper
+     * @return wrapped PathInputResource
+     */
+    PathInputResource wrap(Function<SeekableByteChannel, SeekableByteChannel> wrapper) {
```

Comment:
This had something to do with how GATK uses it. I don't remember now, it has been too long and I lost the context. If this function's a problem then I should also push the GATK PR and we can work them in concert.